### PR TITLE
Remove references to deprecated `iid` in SearchCV.

### PR DESCRIPTION
**Pull request by hristog**
Addresses https://github.com/dask/dask-ml/issues/791.

**tl;dr**
`iid` has been deprecated and the updated behavior corresponds to `iid=True`.

**Additional Information**

Please, note that the original values in `machine-learning/scale-scikit-learn.ipynb` were set to `iid=False`. I couldn't think of a way to communicate the change of behavior in a markdown cell in the notebook, because my understanding is that the change doesn't affect the example to a significant enough degree to warrant an actual explanation.

I've also checked the remaining examples, in case some of the other model selection imports, were affected by the `iid` deprecation (since it was done at the `BaseSearchCV` level).

**References:**
- Deprecation warning and explanation of associated behavior: https://github.com/scikit-learn/scikit-learn/pull/9379.
- Actual deprecation: https://github.com/scikit-learn/scikit-learn/pull/13834.

Thanks!

### CODE REVIEW

**Review by jrbourbeau — APPROVED** [review]
Thanks @hristog!
